lib/rubocop/cop/performance/method_object_as_block.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop identifies places where methods are converted to blocks, with the
+      # use of `&method`, and passed as arguments to `Enumerable` method calls.
+      # It is faster to replace those with explicit blocks, calling those methods inside.
+      #
+      # @example
+      #   # bad
+      #   array.map(&method(:do_something))
+      #
+      #   # good
+      #   array.map { |x| do_something(x) }
+      #
+      class MethodObjectAsBlock < Base
+        MSG = 'Use block explicitly instead of passing a method object.'
+
+        def_node_matcher :method_object?, <<~PATTERN
+          (block-pass (send nil? :method sym))
+        PATTERN
+
+        def on_block_pass(node)
+          method_object?(node) do
+            send_node = node.parent
+            add_offense(node) if send_node&.enumerable_method?
```

Comment:
Why restrict it to enumerable methods?

Reply:
Yeah, I just realized that was my mistake. We should detect `&method` as an argument for any method.